ethereumj-core/src/main/java/org/ethereum/mine/BlockMiner.java

```diff
@@ -265,6 +265,9 @@ protected void restartMining() {
                 MinerIfc localMiner = config.getBlockchainConfig()
                         .getConfigForBlock(miningBlock.getNumber())
                         .getMineAlgorithm(config);
+                if (localMiner instanceof EthashMiner) {
```

Comment:
sorry, but it looks really close to what we had in the first iteration 🤣  Maybe bring back that `setListeners` method to the interface and remove type checking?